### ISSUE TITLE
New image shares working directory of another

### ISSUE THREAD

**Issue by jvalteren** [pharo-project/pharo-launcher#340]
I've noticed some concerning behavior of the Pharo Launcher. I'm currently using version 1.8 (latest downloaded from the pharo.org website) on macOS High Sierra, Version 10.13.6 (17G7024).

I have been working almost exclusively in 1 image over the last weeks. However, every now and then I would want to try something out in a fresh image, so then I use the launcher to create a new (almost always 7.0 64-bit) Pharo image, and do some work in there.

Recently I needed to work towards deployment of a Seaside web application, so I started testing my baseline in a fresh image. However, I noticed some strange behavior, which I've managed to trace to the following:

When using the normal, green Launch button (or double-clicking) on an image, **it starts with the working directory of the image I've been working in before**, over the last couple of weeks.

_To reproduce (hopefully):_
For example, I've been working a lot in an image called `p7-papertrail`. When I now create a new image, give it the name 'p7-launcher-bug' and launch it, the following code... `FileLocator localDirectory asFileReference.`
...produces the result: `File @ /Users/jvalteren/Documents/Pharo/images/p7-papertrail/pharo-local`.
**It should have returned `File @ /Users/jvalteren/Documents/Pharo/images/p7-launcher-bug/pharo-local`.**

This is wrong and poses a serious risk of losing data. I've found out about this because, when testing my baseline, I started removing Iceberg repositories from the file system (using 'Forget repository'). Only to discover that in my main image, it was telling me that it was missing local repository files!

For your information, when launching an image with the grey 'Launch without setting' button, this problem doesn't occur.

**Comment by jvalteren** [comment]
@jecisc, you mentioned: 
> For your comment on the share local directory, it might not be a problem of the launcher but of the settings. If you have exported all Pharo settings, it includes a fix location of a local directory, so all images launched with settings will point to this directory.

I have looked at the launcher settings, but didn't find anything related. Can you tell me what you mean by your 'exported all Pharo settings' comment?

**Comment by jecisc** [comment]
I mean if you stored the setting of one of your Pharo image (not the launcher settings)

**Comment by jvalteren** [comment]
Where can I find those settings?

**Comment by jvalteren** [comment]
Meaning: where can I find those settings on the filesystem? Where does the launcher get them from? I inspected the code the launcher executes and finally it comes down to the OS process command (simplified):
`/Users/myUsername/Documents/Pharo/vms/70-x64/Pharo.app/Contents/MacOS/Pharo /Users/myUsername/Documents/Pharo/images/myImage/myImage.image`

I have found the System > Local directory setting in the Settings Browser. I am assuming that I have clicked the 'Store Settings' button once. However, it is not apparent behavior that every other image launched then takes over those settings, especially since the so-called local directory is then changed.

This effectively changes Iceberg's behavior to 'share repositories between images', but in a very weird and unexpected way.

**Comment by jvalteren** [comment]
PS Do I need to move this issue to the Pharo project itself?

**Comment by svenvc** [comment]
StartupPreferencesLoader preferencesVersionFolder / 'system-settings.ston'

I think this is global

**Comment by jvalteren** [comment]
Yes, I think so too, which is a problem. Go to any image and execute this: `IceLibgitRepository repositoriesLocation.`
It will return the 'iceberg' directory under the System > Local directory configured in the settings.

**Comment by casco** [comment]
I experienced a similar behaviour some weeks ago. macOs High Sierra too. Pharo Image 7.0.3.

Solved it by removing ~/Library/Preferences/pharo

Before getting to that point, I realized that:

The class **SystemResolver** had a old value in its class variable **UserLocalDirectory** . This affects the behaviour of the localDirectory resolution, which in turns affects where iceberg looks for repositories.

Not sure how/when that variable is set (Could not dig any further). But it seems to initialize its value from a Preferences file, if it is present (at least on macOs) .

BTW, it is not an issue with the PharoLauncher per se, but with the way Pharo 7 (I guess) handles preferences. 

Hope this helps.
